From 56f3c94583e2502063d7b503f3725895aec54c82 Mon Sep 17 00:00:00 2001
From: t <tristan75015@gmail.com>
Date: Tue, 20 Oct 2026 01:18:22 +0200
Subject: [PATCH] MFront: messages of the bounds of material properties

- castem interface: a violated physical upper bound was reported as
  "below its physical upper bound";
- generic interface: a violated lower bound with the strict policy was
  only reported as "out of bounds", unlike the other cases.
---
 mfront/src/CastemMaterialPropertyInterface.cxx      | 2 +-
 mfront/src/GenericMaterialPropertyInterfaceBase.cxx | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/mfront/src/CastemMaterialPropertyInterface.cxx b/mfront/src/CastemMaterialPropertyInterface.cxx
index 2e3173329..34bc8ea12 100644
--- a/mfront/src/CastemMaterialPropertyInterface.cxx
+++ b/mfront/src/CastemMaterialPropertyInterface.cxx
@@ -135,7 +135,7 @@ namespace mfront {
       out << "if(" << v.name << " > " << v.type << "(" << b.upperBound
           << ")){\n"
           << "cerr << \"" << name << ": " << v.name
-          << " is below its physical upper bound (\"\n << " << v.name
+          << " is over its physical upper bound (\"\n << " << v.name
           << " << \">" << b.upperBound << ").\\n\";\n"
           << "return std::nan(\"\");\n"
           << "}\n";
diff --git a/mfront/src/GenericMaterialPropertyInterfaceBase.cxx b/mfront/src/GenericMaterialPropertyInterfaceBase.cxx
index 18e73ad38..af3806876 100644
--- a/mfront/src/GenericMaterialPropertyInterfaceBase.cxx
+++ b/mfront/src/GenericMaterialPropertyInterfaceBase.cxx
@@ -175,7 +175,8 @@ namespace mfront {
       os << "if(" << v.name << " < " << v.type << "(" << b.lowerBound << ")){\n"
          << "if(mfront_out_of_bounds_policy==" << iucname
          << "_STRICT_POLICY){\n"
-         << "mfront_report(\"" << v.name << " is out of bounds.\");\n"
+         << "mfront_report(\"" << v.name << " is below its lower bound (\" + "
+         << to_string << " + \"<" << b.lowerBound << ").\\n\");\n"
          << "mfront_output_status->status = -1;\n"
          << "mfront_output_status->bounds_status = -" << i << ";\n"
          << "errno = mfront_errno_old;\n"
-- 
2.43.0

